addons/survey/static/src/js/survey_result.js

```diff
@@ -453,6 +453,7 @@ publicWidget.registry.SurveyResultWidget = publicWidget.Widget.extend({
         'click a.filter-passed-and-failed': '_onFilterPassedAndFailedClick',
         'click .o_survey_answer_image': '_onAnswerImgClick',
         "click .o_survey_results_print": "_onPrintResultsClick",
+        "click [role='tablist'] a": "_onDataViewChange",
```

Comment:
Would maybe add a specific class on that "a" to avoid a complex selector that could be easily broken.